Src/ILGPU.Tests/AtomicCASOperations.tt

```diff
@@ -7,7 +7,7 @@ using Xunit;
 using Xunit.Abstractions;
 
 <#
-var atomicTypes = AtomicNumericTypes;
+var atomicTypes = DefaultIntTypes.Concat(FloatTypes.Skip(1)).ToArray();
```

Comment:
Do you suggest that we should change the definition of the `AtomicNumericTypes` array? I am asking because we should not generate `16bit` atomics at the moment.

Reply:
`AtomicNumericTypes` appears to be equivalent to `DefaultIntTypes.Concat(FloatTypes.Skip(1))`, containing the same primitive types. I was thinking it would be easier to use `AtomicNumericTypes`, and if we added 16-bit atomics, we would just have to update the one spot (`AtomicNumericTypes`).

Reply:
Hmmm.. looking at the definition of `AtomicNumericTypes`, we might want to clean it up.

The class `AtomicTypeInformation` does not provide any additional functionality over the base class `TypeInformation`. In addition, the atomic definition for `float` does not define the `f` suffix.

Reply:
I am going to polish up the class definition and the `AtomicNumericTypes` array.